### ISSUE TITLE
Bug: no-unused-vars - 'x' is assigned a value but only used as a type

### ISSUE THREAD

**Issue by burt202** [typescript-eslint/typescript-eslint#9778]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.2&fileType=.tsx&code=MYewdgzgLgBFCm0YF4YG8YDMQgFwwEYAmAZhgF8AoSqATwAd4YAxHFOB%2BETORKa0JFggARgCt8rEOzTY8hKpUEQQAG3gA6VSADmACgBEosQYA0MYwEogA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgyRE0aB2iRwYAL4hlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const test = { foo: 123 }

type Foo = typeof test

const obj: Foo = {foo: 1}

console.log("obj", obj)
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-unused-vars": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

In v7, the code does not report any error, but in v8 you get `'test' is assigned a value but only used as a type. 1:7 - 1:11`

Code samples above taken from playground example using `v5.5.2` of TS and `v9.3.0` of eslint

### Actual Result

Its errors with `'test' is assigned a value but only used as a type. 1:7 - 1:11`.

I see there has been some changes to the `no-unused-vars` rule for v8 but i wondered if there was an escape hatch for this error because i really dont want to have to ignore this rule for the line etc??

**Comment by bradzacher** [comment]
This is fully expected.
You have created a runtime value and only used it in a type. Thus the runtime value is not used.

If you intended to create an unused runtime value then you should use a disable comment and add a comment explaining why you're using a runtime value insrsad of just constructing the type using types.

**Comment by burt202** [comment]
Ok thanks for the clarification
